client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -10,34 +10,43 @@ import { Button } from '@wordpress/components';
  * Internal dependencies
  */
 import { STORE_KEY } from '../stores/onboard';
+import DesignSelector from './design-selector';
 import StepperWizard from './stepper-wizard';
 import VerticalSelect from './vertical-select';
 import SiteTitle from './site-title';
+import { Steps } from '../types';
 import './style.scss';
 
-export default function OnboardingEdit() {
+export default function OnboardingEdit( { attributes: { step = 0 } } ) {
```

Comment:
Adding types to this component will get us autocomplete when adding other cases and also, in current situation, a type error:
`Type 'Steps.DesignSelection' is not comparable to type 'Steps.IntentGathering'.`

```suggestion
import { Attributes } from '.';

interface OnboardingEditProps {
	attributes: Attributes;
}

const OnboardingEdit: FunctionComponent< OnboardingEditProps > = ( { attributes: { step } } ) => {```

Reply:
4246ca6d86